pulp_rpm/tests/functional/utils.py

```diff
@@ -32,17 +33,15 @@ def set_up_module():
     require_pulp_plugins({'pulp_rpm'}, SkipTest)
 
 
-def gen_rpm_remote(**kwargs):
-    """Return a semi-random dict for use in creating a rpm Remote.
+def gen_rpm_remote(url=None, **kwargs):
+    """Return a semi-random dict for use in creating a RPM remote.
 
     :param url: The URL of an external content source.
     """
-    remote = gen_remote(RPM_SIGNED_FIXTURE_URL)
-    rpm_extra_fields = {
-        **kwargs
-    }
-    remote.update(rpm_extra_fields)
-    return remote
+    if url is None:
+        url = RPM_UNSIGNED_FIXTURE_URL
```

Comment:
This is perfectly fine, and if you prefer it this way I'm +1.  But just FWIW the reason you would typically do this over defining a kwarg default value is if you've got a mutable data structure like a list or a dict, but strings are immutable in Python so there's no need to do it for strings.

Also is there a particular reason to default to unsigned RPMs vs signed RPMs? No objection here either though.

Reply:
I just want to have the same standards used across tests. I do not mind change here. But what we adopt for one plugin, I would like to the same to adopt to all the others.

I adopted the same one used to [file_plugin](https://github.com/pulp/pulp_file/blob/bfe6eb2e76a796a686f0c0260dc4c51155fa2667/pulp_file/tests/functional/utils.py#L53)

```python
def gen_file_remote(url=None, **kwargs):
    """Return a semi-random dict for use in creating a file Remote.
    :param url: The URL of an external content source.
    """
    if url is None:
        url = FILE_FIXTURE_MANIFEST_URL

return gen_remote(url, **kwargs)
```

Unsigned RPMs are parsed faster.  If the test does not explicit requires signed RPMs, there is no need to use them. We adopted unsigned RPMs for most of the tests cases for Pulp 2.

Reply:
+1 x2

Reply:
`url = url or FILE_FIXTURE_MANIFEST_URL` is an option to this :)